plugins/modules/sda_fabric_devices_workflow_manager.py

```diff
@@ -4022,6 +4022,55 @@ def get_diff_merged(self, config):
 
         return self
 
+    def arrange_config_for_deletion(self, device_config):
+        """
+        Arrange the device config in such a way that the device with the role 'CONTROL_PLANE_NODE'
+        will be deleted at last.
+
+        Parameters:
+            device_config (list): Devices config details provided by the user in the playbook.
+        Returns:
+            updated_device_config (list): Updated device config details.
+        Description:
+            Loop in the device config information. If the device doesnot exists, prepend the
+            config details and the information collected from the Cisco Catalyst Center to the
+            new list or if the device has a role of 'CONTROL_PLANE_NODE', append the config details
+            and the information collected from the Cisco Catalyst Center to the new list.
+            Update the self.have.get("fabric_details") and return the new device_config list.
+        """
+
+        fabric_device_index = -1
+        updated_device_config = []
+        update_have = []
+        for item in device_config:
+            fabric_device_index += 1
+            device_ip = item.get("device_ip")
+            have_device_details = self.have.get("fabric_devices")[fabric_device_index]
+            exists = have_device_details.get("exists")
+            if not exists:
+                self.log(
+                    "The device with IP address '{ip}' is not available in the Cisco Catalyst Center."
+                    .format(ip=device_ip)
+                )
+                updated_device_config = [item] + updated_device_config
+                update_have = [have_device_details] + update_have
+                continue
+
+            device_roles = have_device_details.get("device_details").get("deviceRoles")
+            if "CONTROL_PLANE_NODE" in device_roles:
```

Comment:
self.log("Device with IP '{}' has role 'CONTROL_PLANE_NODE', appending to the end.".format(device_ip))